com.unity.netcode.gameobjects/Runtime/Interest/InterestManager.cs

```diff
@@ -60,19 +56,24 @@ public void AddObject(ref TObject obj)
 
         public void AddDefaultInterestNode(TObject obj)
         {
-            obj.AddInterestNode(m_DefaultInterestNode);
+            AddInterestNode(ref obj, m_DefaultInterestNode);
         }
 
         public void RemoveObject(ref TObject obj)
         {
-            List<IInterestNode<TObject>> nodes = obj.GetInterestNodes();
-            foreach (var node in nodes)
+            if (m_InterestNodesMap.TryGetValue(obj, out var nodes))
             {
-                if (node == null)
+                foreach (var node in nodes)
                 {
-                    continue;
+                    if (node == null)
+                    {
+                        continue;
+                    }
+
+                    node.RemoveObject(obj);
                 }
-                node.RemoveObject(obj);
+
+                m_InterestNodesMap.Remove(obj);
```

Comment:
This should fix the issue @NoelStephensUnity  @MFatihMAR @jeffreyrainy found w/ pooling